packages/allure-playwright/src/index.ts

```diff
@@ -17,130 +18,160 @@
+    const allureTest = this.allureTestCache.get(test);
+    if (!allureTest) {
+      return;
+    }
+    allureTest.status = statusToAllureStats(result.status, test.expectedStatus);
+    if (result.error) {
+      const message = result.error.message && stripAscii(result.error.message);
+      let trace = result.error.stack && stripAscii(result.error.stack);
+      if (trace && message && trace.startsWith(`Error: ${message}`)) {
+        trace = trace.substr(message.length + "Error: ".length);
+      }
+      allureTest.statusDetails = {
+        message,
+        trace,
+      };
+    }
+    for (const attachment of result.attachments) {
+      if (!attachment.body && !attachment.path) {
+        continue;
+      }
+
+      let fileName;
+      if (attachment.body) {
+        fileName = runtime.writeAttachment(attachment.body, attachment.contentType);
+      } else {
+        if (!fs.existsSync(attachment.path!)) {
+          continue;
         }
+        fileName = runtime.writeAttachmentFromPath(attachment.path!, attachment.contentType);
       }
-      for (const child of suite.suites) {
-        processSuite(child, group);
+      allureTest.addAttachment(attachment.name, attachment.contentType, fileName);
+      if (attachment.name === "diff") {
+        allureTest.addLabel("testType", "screenshotDiff");
       }
-      group.endGroup();
-    };
-    processSuite(this.suite, runtime);
+    }
+
+    for (const stdout of result.stdout) {
+      allureTest.addAttachment(
+        "stdout",
+        "text/plain",
+        runtime.writeAttachment(stdout, "text/plain"),
+      );
+    }
+    for (const stderr of result.stderr) {
+      allureTest.addAttachment(
+        "stderr",
+        "text/plain",
+        runtime.writeAttachment(stderr, "text/plain"),
+      );
+    }
+    allureTest.endTest();
+  }
 
+  onEnd(): void {
+    for (const group of this.allureGroupCache.values()) {
+      group.endGroup();
+    }
     if (process.env.PW_ALLURE_POST_PROCESSOR_FOR_TEST) {
       try {
+        const writer = this.allureWriter;
+        void writer; // Used in `eval()`, below.
         const postProcess = eval(process.env.PW_ALLURE_POST_PROCESSOR_FOR_TEST); // eslint-disable-line no-eval
-        console.log(JSON.stringify(postProcess(writerForTest))); // eslint-disable-line no-console
+        console.log(JSON.stringify(postProcess(this.allureWriter))); // eslint-disable-line no-console
       } catch (e) {
-        console.log(JSON.stringify({ error: (e as Error).toString() })); // eslint-disable-line no-console
+        console.log(JSON.stringify({ error: (e as Error).stack || String(e) })); // eslint-disable-line no-console
       }
     }
   }
+
+  private getAllureGroup(suite: Suite): AllureGroup {
+    let group = this.allureGroupCache.get(suite);
+    if (!group) {
+      const parent = suite.parent ? this.getAllureGroup(suite.parent) : this.allureRuntime;
+      group = parent.startGroup(suite.title);
```

Comment:
It is not expected that `get*` methods have any side effects. Better to call it `ensureGroupStarted`